src/Checkbox/CheckboxGroup.js

```diff
@@ -0,0 +1,47 @@
+// @flow weak
+
+import React, { PropTypes, Children } from 'react';
+import { createStyleSheet } from 'jss-theme-reactor';
+import classNames from 'classnames';
+
+export const styleSheet = createStyleSheet('CheckboxGroup', () => {
+  return {
+    root: {
+      display: 'flex',
+      flexDirection: 'row',
+    },
+    label: {
+      marginRight: 16,
+    },
+  };
+});
+
+/**
+ * Primary purpose of the CheckboxGroup is to display checkboxes in a compact row layout to conform with
+ *  the spec goal of preserving space.
+ *
+ *  @see https://material.google.com/components/selection-controls.html#selection-controls-checkbox
+ */
+export default function CheckboxGroup(props, context) {
```

Comment:
Looking at the implementation, I think that it could be a `FormGroup` component that we would use instead of a `CheckboxGroup` and `SwitchGroup`.